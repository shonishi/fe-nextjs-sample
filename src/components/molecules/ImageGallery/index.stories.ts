import type { Meta, StoryObj } from '@storybook/react';
import ImageGallery from '.';

const meta: Meta<typeof ImageGallery> = {
  component: ImageGallery,
};

export default meta;
type Story = StoryObj<typeof ImageGallery>;

export const MultipleElement: Story = {
  args: {
    images: [
      {
        src: 'https://www.pakutaso.com/shared/img/thumb/PAK74_ahirucyansusumu_TP_V.jpg',
        alt: 'Two each of gray, white, and black shirts laying flat.',
      },
      {
        src: 'https://www.pakutaso.com/shared/img/thumb/PAK85_enjinahirucyan_TP_V.jpg',
        alt: 'Model wearing plain black basic tee.',
      },
      {
        src: 'https://www.pakutaso.com/shared/img/thumb/PAK88_amazonst_TP_V.jpg',
        alt: 'Model wearing plain gray basic tee.',
      },
      {
        src: 'https://www.pakutaso.com/shared/img/thumb/NKJ56_dateahiru_TP_V.jpg',
        alt: 'Model wearing plain white basic tee.',
      },
    ],
  },
};

export const SingleElement: Story = {
  args: {
    images: [
      {
        src: 'https://www.pakutaso.com/shared/img/thumb/PAK74_ahirucyansusumu_TP_V.jpg',
        alt: 'Two each of gray, white, and black shirts laying flat.',
      },
    ],
  },
};

export const EmptyElement: Story = {
  args: {
    images: [],
  },
};
