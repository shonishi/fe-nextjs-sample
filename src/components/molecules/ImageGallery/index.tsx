import NextImage from 'next/image';

interface Props {
  images: { src: string; alt: string }[];
}

export default function ImageGallery({ images }: Props) {
  return (
    <div className="mx-auto mt-6 max-w-2xl sm:px-6 lg:grid lg:max-w-7xl lg:grid-cols-3 lg:gap-x-8 lg:px-8">
      <div className="aspect-h-4 aspect-w-3 hidden overflow-hidden rounded-lg lg:block">
        <NextImage
          src={images[0].src}
          alt={images[0].alt}
          className="h-full w-full object-cover object-center"
          fill
        />
      </div>
      {images.length > 1 && (
        <div className="hidden lg:grid lg:grid-cols-1 lg:gap-y-8">
          <div className="aspect-h-2 aspect-w-3 overflow-hidden rounded-lg">
            <NextImage
              src={images[1].src}
              alt={images[1].alt}
              className="h-full w-full object-cover object-center"
              fill
            />
          </div>
          {images.length > 2 && (
            <div className="aspect-h-2 aspect-w-3 overflow-hidden rounded-lg">
              <NextImage
                src={images[2].src}
                alt={images[2].alt}
                className="h-full w-full object-cover object-center"
                fill
              />
            </div>
          )}
        </div>
      )}
      {images.length > 3 && (
        <div className="aspect-h-5 aspect-w-4 lg:aspect-h-4 lg:aspect-w-3 sm:overflow-hidden sm:rounded-lg">
          <NextImage
            src={images[3].src}
            alt={images[3].alt}
            className="h-full w-full object-cover object-center"
            fill
          />
        </div>
      )}
    </div>
  );
}
