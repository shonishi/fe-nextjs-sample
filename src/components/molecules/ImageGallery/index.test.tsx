import ImageGallery from '.';
import renderer from 'react-test-renderer';

describe('ImageGallery', () => {
  it('imagesの要素が複数の場合、全て表示されること', async () => {
    const tree = renderer
      .create(
        <ImageGallery
          {...{
            images: [
              {
                src: 'https://www.pakutaso.com/shared/img/thumb/PAK74_ahirucyansusumu_TP_V.jpg',
                alt: 'Two each of gray, white, and black shirts laying flat.',
              },
              {
                src: 'https://www.pakutaso.com/shared/img/thumb/PAK85_enjinahirucyan_TP_V.jpg',
                alt: 'Model wearing plain black basic tee.',
              },
              {
                src: 'https://www.pakutaso.com/shared/img/thumb/PAK88_amazonst_TP_V.jpg',
                alt: 'Model wearing plain gray basic tee.',
              },
              {
                src: 'https://www.pakutaso.com/shared/img/thumb/NKJ56_dateahiru_TP_V.jpg',
                alt: 'Model wearing plain white basic tee.',
              },
            ],
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('imagesの要素が1つの場合、1つだけ表示されること', async () => {
    const tree = renderer
      .create(
        <ImageGallery
          {...{
            images: [
              {
                src: 'https://www.pakutaso.com/shared/img/thumb/PAK74_ahirucyansusumu_TP_V.jpg',
                alt: 'Two each of gray, white, and black shirts laying flat.',
              },
            ],
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('imagesが空の場合、何も表示されないこと', async () => {
    const tree = renderer.create(<ImageGallery {...{ images: [] }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
