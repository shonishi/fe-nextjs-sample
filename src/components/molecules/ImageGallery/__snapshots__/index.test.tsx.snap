// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGallery imagesが空の場合、何も表示されないこと 1`] = `null`;

exports[`ImageGallery imagesの要素が1つの場合、1つだけ表示されること 1`] = `
<div
  className="mx-auto mt-6 max-w-2xl sm:px-6 lg:grid lg:max-w-7xl lg:grid-cols-3 lg:gap-x-8 lg:px-8"
>
  <div
    className="aspect-h-4 aspect-w-3 hidden overflow-hidden rounded-lg lg:block"
  >
    <img
      alt="Two each of gray, white, and black shirts laying flat."
      className="h-full w-full object-cover object-center"
      data-nimg="1"
      decoding="async"
      height={0}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=3840&q=75 3840w"
      style={
        {
          "color": "transparent",
        }
      }
      width={0}
    />
  </div>
</div>
`;

exports[`ImageGallery imagesの要素が複数の場合、全て表示されること 1`] = `
<div
  className="mx-auto mt-6 max-w-2xl sm:px-6 lg:grid lg:max-w-7xl lg:grid-cols-3 lg:gap-x-8 lg:px-8"
>
  <div
    className="aspect-h-4 aspect-w-3 hidden overflow-hidden rounded-lg lg:block"
  >
    <img
      alt="Two each of gray, white, and black shirts laying flat."
      className="h-full w-full object-cover object-center"
      data-nimg="1"
      decoding="async"
      height={0}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK74_ahirucyansusumu_TP_V.jpg&w=3840&q=75 3840w"
      style={
        {
          "color": "transparent",
        }
      }
      width={0}
    />
  </div>
  <div
    className="hidden lg:grid lg:grid-cols-1 lg:gap-y-8"
  >
    <div
      className="aspect-h-2 aspect-w-3 overflow-hidden rounded-lg"
    >
      <img
        alt="Model wearing plain black basic tee."
        className="h-full w-full object-cover object-center"
        data-nimg="1"
        decoding="async"
        height={0}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=3840&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK85_enjinahirucyan_TP_V.jpg&w=3840&q=75 3840w"
        style={
          {
            "color": "transparent",
          }
        }
        width={0}
      />
    </div>
    <div
      className="aspect-h-2 aspect-w-3 overflow-hidden rounded-lg"
    >
      <img
        alt="Model wearing plain gray basic tee."
        className="h-full w-full object-cover object-center"
        data-nimg="1"
        decoding="async"
        height={0}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=3840&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FPAK88_amazonst_TP_V.jpg&w=3840&q=75 3840w"
        style={
          {
            "color": "transparent",
          }
        }
        width={0}
      />
    </div>
  </div>
  <div
    className="aspect-h-5 aspect-w-4 lg:aspect-h-4 lg:aspect-w-3 sm:overflow-hidden sm:rounded-lg"
  >
    <img
      alt="Model wearing plain white basic tee."
      className="h-full w-full object-cover object-center"
      data-nimg="1"
      decoding="async"
      height={0}
      onError={[Function]}
      onLoad={[Function]}
      sizes="100vw"
      src="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fwww.pakutaso.com%2Fshared%2Fimg%2Fthumb%2FNKJ56_dateahiru_TP_V.jpg&w=3840&q=75 3840w"
      style={
        {
          "color": "transparent",
        }
      }
      width={0}
    />
  </div>
</div>
`;
